packages/test/test-end-to-end-tests/src/test/signalPerfTelemetry.spec.ts

```diff
@@ -0,0 +1,367 @@
+ * Licensed under the MIT License.
+ */
+
+import { describeCompat, itExpects } from "@fluid-private/test-version-utils";
+import { IContainer } from "@fluidframework/container-definitions/internal";
+import { ConnectionState } from "@fluidframework/container-loader";
+import { MockLogger } from "@fluidframework/telemetry-utils/internal";
+import {
+	DataObjectFactoryType,
+	ITestContainerConfig,
+	ITestFluidObject,
+	ITestObjectProvider,
+	getContainerEntryPointBackCompat,
+	timeoutPromise,
+} from "@fluidframework/test-utils/internal";
+
+const waitForSignal = async (...signallers: { once(e: "signal", l: () => void): void }[]) =>
+	Promise.all(
+		signallers.map(async (signaller, index) =>
+			timeoutPromise((resolve) => signaller.once("signal", () => resolve()), {
+				durationMs: 2000,
+				errorMsg: `Signaller[${index}] Timeout`,
+			}),
+		),
+	);
+
+describeCompat("Signal performance telemetry", "NoCompat", (getTestObjectProvider, apis) => {
+	let provider: ITestObjectProvider;
+	let dataObject: ITestFluidObject;
+	let container: IContainer;
+	let containerRuntime;
+	let logger: MockLogger;
+
+	beforeEach("getTestObjectProvider", async () => {
+		provider = getTestObjectProvider();
+
+		logger = new MockLogger();
+
+		const testContainerConfig: ITestContainerConfig = {
+			fluidDataObjectType: DataObjectFactoryType.Test,
+			loaderProps: {
+				logger,
+			},
+		};
+		container = await provider.makeTestContainer(testContainerConfig);
+		dataObject = await getContainerEntryPointBackCompat<ITestFluidObject>(container);
+		containerRuntime = dataObject.context.containerRuntime;
+
+		// need to be connected to send signals
+		if (container.connectionState !== ConnectionState.Connected) {
+			await new Promise((resolve) => container.once("connected", resolve));
+		}
+	});
+
+	describe("for all drivers", () => {
+		itExpects(
+			"SignalLatency telemetry after 100 broadcast signals",
+			[
+				{
+					eventName: "fluid:telemetry:ContainerRuntime:SignalLatency",
+					clientType: "interactive",
+				},
+			],
+			async () => {
+				// Send 100+ broadcast signals to trigger latency telemetry event
+				for (let i = 0; i < 101; i++) {
+					containerRuntime.submitSignal("signal", "test");
+					await waitForSignal(containerRuntime);
+				}
+			},
+		);
+
+		itExpects(
+			"SignalLost error event after missing signal is detected",
+			[
+				{
+					eventName: "fluid:telemetry:ContainerRuntime:SignalLatency",
+					clientType: "interactive",
+				},
+				{
+					eventName: "fluid:telemetry:ContainerRuntime:SignalLost",
+					clientType: "interactive",
+				},
+			],
+			async () => {
+				for (let i = 0; i < 101; i++) {
+					containerRuntime.submitSignal("signal", "test");
+					await waitForSignal(containerRuntime);
+				}
+
+				// Process a signal with a sequence number that is higher than expected to simulate missing signals and trigger a SignalLost error event.
+				containerRuntime.processSignal(
```

Comment:
Looks like we are faking sending a response for a signal that was never sent out.
It doesn't simulate a real situation that we ever expect to be in.
For good testing we client to send signals and then we control the responses. So, client sends for example 3 signals and then we respond by dropping the second.

Reply:
Went with an alternate approach using a test stub for 'processSignal'. This allows us to simulate a dropped signal by altering the behavior of the response to the sent signal.